kroma-validator/l2_output_submitter.go

```diff
@@ -268,30 +298,91 @@ func (l *L2OutputSubmitter) CalculateWaitTime(ctx context.Context, nextBlockNumb
 	return 0
 }
 
-// HasEnoughDeposit checks if validator has enough deposit to bond when trying output submission.
-func (l *L2OutputSubmitter) HasEnoughDeposit(ctx context.Context) (bool, error) {
+// CanSubmitOutput checks if the validator satisfies the condition to submit L2Output.
+func (l *L2OutputSubmitter) CanSubmitOutput(ctx context.Context, outputIndex *big.Int) (bool, error) {
 	cCtx, cCancel := context.WithTimeout(ctx, l.cfg.NetworkTimeout)
 	defer cCancel()
 	from := l.cfg.TxManager.From()
-	balance, err := l.valpoolContract.BalanceOf(optsutils.NewSimpleCallOpts(cCtx), from)
-	if err != nil {
-		return false, fmt.Errorf("failed to fetch deposit amount: %w", err)
+	if l.IsValPoolTerminated(outputIndex) {
+		validatorStatus, err := l.getValidatorStatus(ctx)
+		if err != nil {
+			return false, err
+		}
+
+		validatorActivated, err := l.tryActivateValidator(validatorStatus)
+		if err != nil {
+			return false, err
+		}
+		if !validatorActivated {
+			return false, nil
+		}
+
+		if validatorStatus != StatusActive {
+			l.log.Warn("validator has activated, but currently has not enough tokens", "currentStatus", validatorStatus)
+			return false, nil
+		}
+
+		if isInJail, err := l.isInJail(ctx); err != nil {
+			return false, err
+		} else if isInJail {
+			l.log.Warn("validator is in jail")
+			return false, nil
+		}
```

Comment:
Once we get rid of `tryActivateValidator` in favor of initializing it, this part will be simpler. 
We only need to check that validatorStatus is `ACTIVE` and not in jail.

Reply:
Hmm maybe it's better to remove `tryActivateValidator` at all. Since this transaction is quite expensive, it should be done by the validator manually, not automatically.
If we remain it in code, I think it needs to be done in loop, not initializing. Because validator status can be varied during running the client.

Reply:
I'm in favor of making it manual for the validator to do, I think a well-written guide for commands for register and activate would be fine.

Reply:
This was to let a validator submit outputs automatically if he gets more amounts delegated tokens than `MIN_ACTIVATE_AMOUNT`.  I think it would become simpler with your comment above, thanks!

Reply:
Also agree with the point that this action is expensive and we should give them opportunity to manually do this. Then maybe I can move this to #302 and add an extra subcommand for activating validator!